src/main/java/com/acme/bank/investmentservice/service/EligibilityService.java

```diff
@@ -1,47 +1,52 @@
 package com.acme.bank.investmentservice.service;
 
 import com.acme.bank.investmentservice.domain.UserProfile;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
-import java.util.HashMap;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
 
 @Service
 public class EligibilityService {
 
-    @Autowired
-    private UserProfileService userProfileService;
+    public EligibilityService(UserProfileService userProfileService) {
+        this.userProfileService = userProfileService;
+    }
+
+    private static final float DEPOSIT_BALANCE_REQUIREMENT_FOR_ELDERLY = 1_000_000f;
+    private static final float DEPOSIT_BALANCE_REQUIREMENT_FOR_OTHERS = 500_000f;
+    private final UserProfileService userProfileService;
 
-    // UserProfile retrieval is an intensive call, hence we want to cache the operation
-    // No need to consider cache invalidation for simplicity
-    private HashMap<String, UserProfile> cache;
+    private final ConcurrentHashMap<String, UserProfile> userProfileCache = new ConcurrentHashMap<>();
```

Comment:
Spring bean is singleton, meaning that and instance variables are not thread safe